client/src/components/AuditAdmin/Setup/Contests/ContestForm.tsx

```diff
@@ -292,7 +294,15 @@ const ContestForm: React.FC<IProps> = ({
         setFieldValue,
         isSubmitting,
       }: FormikProps<{ contests: IContestValues[] }>) => (
-        <form data-testid="form-one">
+        <form
+          data-testid="form-one"
+          style={{ width: '100%' }}
+          onSubmit={e => {
+            e.preventDefault()
+            if (isOpportunisticFormClean(touched, values)) goToNextStage()
```

Comment:
I found it a bit unintuitive that `isOpportunisticFormClean` could be called even when not dealing with an opportunistic form. I think I'd prefer the check for whether the form is for target vs. opportunistic contests to be separated from the check for whether the form is clean, e.g.

```
const isOpportunistic = !isTargeted;
if (isOpportunistic && isFormClean(touched, values)) goToNextStage();
```

Separately, I'm sure I'm missing something simple but why can't we call `goToNextStage()` if the form is clean but for target contests?

Reply:
This was existing code that I just moved around. I'm pretty sure it's buggy. The whole architecture for the setup stepper needs to be redone. It's too brittle for me to even risk editing it.

Reply:
Ah missed that it was existing code! Def fine to leave as is then